include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -336,6 +339,155 @@ class gap_decorator
+        * \{
+        */
+        //!\brief Pre-increment, returns updated iterator.
+        constexpr gap_decorator_iterator_ra & operator++() noexcept
+        {
+            base_t::operator++();
+            return *this;
+        }
+
+        //!\brief Pre-decrement, returns updated iterator.
+        constexpr gap_decorator_iterator_ra & operator--() noexcept
+        {
+            base_t::operator--();
+            return *this;
+        }
+
+        //!\brief Post-increment, returns previous iterator state (delegates to pre-increment).
+        constexpr gap_decorator_iterator_ra operator++(int) noexcept
+        {
+            gap_decorator_iterator_ra cpy{*this};
+            ++(*this);
+            return cpy;
+        }
+
+        //!\brief Post-decrement, returns previous iterator state (delegates to pre-decrement).
+        constexpr gap_decorator_iterator_ra operator--(int) noexcept
+        {
+            gap_decorator_iterator_ra cpy{*this};
+            --(*this);
+            return cpy;
+        }
+
+        //!\brief Forward this iterator.
+        constexpr gap_decorator_iterator_ra & operator+=(difference_type const skip) noexcept
+        {
+            this->jump(this->pos + skip);
+            return *this;
+        }
+
+        //!\brief Forward copy of this iterator.
+        constexpr gap_decorator_iterator_ra operator+(difference_type const skip) const noexcept
+        {
+            return gap_decorator_iterator_ra{*(this->host), this->pos + skip};
+        }
+
+        //!\brief Non-member operator+ delegates to non-friend operator+.
+        constexpr friend gap_decorator_iterator_ra operator+(difference_type const skip,
+                                                             gap_decorator_iterator_ra const & it) noexcept
+        {
+            return it + skip;
+        }
+
+        //!\brief Decrement iterator by skip.
+        constexpr gap_decorator_iterator_ra & operator-=(difference_type const skip) noexcept
+        {
+            this->jump(this->pos - skip);
+            return *this;
+        }
+
+        //!\brief Return decremented copy of this iterator.
+        constexpr gap_decorator_iterator_ra operator-(difference_type const skip) const noexcept
+        {
+            return gap_decorator_iterator_ra{*(this->host), this->pos - skip};
+        }
+
+        //!\brief Non-member operator- delegates to non-friend operator-.
+        constexpr friend gap_decorator_iterator_ra operator-(difference_type const skip,
+                                                             gap_decorator_iterator_ra const & it) noexcept
+        {
+            return it - skip;
+        }
+
+        //!\brief Returns the distance between two iterators.
+        constexpr difference_type operator-(gap_decorator_iterator_ra const lhs) const noexcept
+        {
+            return static_cast<difference_type>(this->pos - lhs.pos);
+        }
+        //!\}
+
+        /*!\name Element access
```

Comment:
can you check if this looks fine in the docu? I had a lot of trouble getting named groups being merged if they are repeated in the derived class. For me it always added a second group with the same name. It might help to make a using declaration of the *-operator or a SEQAN3_DOXYGEN_ONLY expression😞

Reply:
with the SEQAN3_DOXYGEN_ONLY and  even without, doxygen does not put them in the same group but makes two group with the same name. I think this is a general bug. Dies the SEQAN3_DOXYGEN_ONLY hack work sometimes?

Reply:
Well it is not a bug. It is simply a missing feature of doxygen. Since you need the base class as a standalone class you won't get around this i guess. I thought maybe he sees that if the variable was overloaded from the base class it will not show the base class version. But it seems this is not the case then?

Reply:
No :( I tried it with and without SEQAN3_DOXYGEN_ONLY